Tests/SourceControlTests/RepositoryManagerTests.swift

```diff
@@ -328,8 +328,6 @@ class RepositoryManagerTests: XCTestCase {
         let variants: [RepositorySpecifier] = [
             .init(url: "https://scm.com/org/foo"),
             .init(url: "https://scm.com/org/foo.git"),
-            .init(url: "http://scm.com/org/foo"),
-            .init(url: "http://scm.com/org/foo.git")
```

Comment:
is there a specific reason for removing these cases, or do they regress with this change applied and are removed because of that?

Reply:
Yah, this doesn't work anymore if we take the protocol as a differentiator. 

IMHO, it was never correct to consider these the same for the purpose of fetching dependencies, it should be possible for a user to switch the protocol if they chose to.

Reply:
> it should be possible for a user to switch the protocol if they chose to.

that is true, but they can still be considered canonical locations. so it is more about how / where use canonical location  than changing the semantic meaning of what canonical location is. to that end, we should probably make a narrower change that does not change CanonicalLocation, but instead changes where / how it is used

Reply:
Right, this is a somewhat narrow change (I need to handle the equality bit that I didn't realize mattered), but this test will have to change since it tests what we're changing.

Reply:
yes, if we are taking about `RepositorySpecifier::storagePath` than these URLs should yield different arenas